core/model/transaction.cpp

```diff
@@ -0,0 +1,103 @@
+
+
+#include "transaction.hpp"
+
+namespace transaction{
+    using command::Transfer;
+    using command::Add;
+
+    template <>
+    Transaction<Transfer<object::Asset>>::Transaction(
+        Object obj
+    ):
+        Transfer(
+            obj
+        )
+    {
+        senderPubkey = obj.dictSub["senderPublicKey"].str;
+        hash = obj.dictSub["hash"].str;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'hash' is assigned in constructor body. Consider performing initialization in initialization list.](https://www.codacy.com/app/soramitsu/iroha/file/4478583020/issues/source?bid=3712491&fileBranchId=3923843#l18)